### PR TITLE
[Lovelace] Add iframe card

### PR DESCRIPTION
**Pull request by covrig**
unbenannt

```yaml
  - type: iframe
    url: https://embed.windy.com/embed2.html?lat=48.234&lon=8.598&zoom=5&level=surface&overlay=radar&menu=&message=&marker=&calendar=&pressure=&type=map&location=coordinates&detail=&detailLat=48.234&detailLon=8.598&metricWind=default&metricTemp=default&radarRange=-12
    aspect_ratio: 50% # optional
```

### CODE REVIEW

**Comment by arsaboo** [comment]
Why do you have the wire-frame card in the screenshot? Don't think that is part of the iframe card.

**Comment by covrig** [comment]
Just for context. They are not related.

**Comment by arsaboo** [comment]
Remove the other card....it is confusing :)

**Comment by c727** [comment]
- we should use the same card layout that we use in the other cards (with title) - remove all the title configs, <-- maybe we create a new element with all the CSS for ha-card that we apply each time? cc: @balloob 
- add refresh interval
- iframe height should be auto-updated

File: src/panels/lovelace/cards/hui-iframe-card.js
Comment (c727) [inline src/panels/lovelace/cards/hui-iframe-card.js]: remove

File: src/panels/lovelace/cards/hui-iframe-card.js
Comment (c727) [inline src/panels/lovelace/cards/hui-iframe-card.js]: `HuiIframeCard `

**Comment by covrig** [comment]
@c727 I cleaned the code. I removed the title feature completely, although I would say it is a nice addition. However, I agree it isn't in line with the rest of the cards.
For the refresh how should I proceed? JavaScript timeout or something else? I don't know if timeouts play well with everything else.
Regarding the height I would keep it like this since it provides a bit more flexibility for the user. The alternative is a bunch of CSS+JavaScript tricks that might or might not work depending on the case.. iframes are finicky.

**Comment by covrig** [comment]
@c727 I updated the title thing as in your drawing. 
Now, thinking a bit more on how I can solve the height issue and other things... it gets rather tricky: scaling issues, screen resolution, can't control the iframe content, no width auto scaling etc. 
However, do we need to care about all this? Nobody is going to embed websites or things that don't auto scale the width. They will stop at widgets, maps, weather and similar things that are specifically designed to be embedded.
If yes, unless somebody else has more time and more experience on how to fix the above-mentioned issues, I would say to keep this as a custom card. If only the height is a issue I can solve it, but it still requires a manual height option for some cases.

---
Another option is to create a [Windy](https://www.windy.com) card where the user can set the location, weather layer, zoom, units, forecast details and so on. Most people only care about this.

**Comment by c727** [comment]
Ok, you are right... Height and refresh are problematic with cross-domain

this is what we have now:
unbenannt

**Comment by covrig** [comment]
So, do I close this and leave it as a custom card? 

Beside the height I can introduce a [scale](https://github.com/covrig/homeassistant-iframe-card/blob/master/www/custom_ui/state-card-iframe.html) factor that will help a lot, but it will still require some trial and error from the user. However, with Lovelace that is easy since they don't need to restart everything. And they only need to do it once.

**Comment by c727** [comment]
No, wait for feedback/ approval

**Comment by c727** [comment]
But yes, a scale factor also sounds nice :+1:

**Comment by covrig** [comment]
@c727 If you have time to work on this it would be great. I can't do anything for the next two weeks (holiday). Otherwise I will gladly continue once I am back. I think we also need some scaling for the content of the iframe. Aspect ration won't be enough for some type of content.

**Review by balloob — APPROVED** [review]

**Comment by covrig** [comment]
@c727 @balloob Thanks for your help. I appreciate it. I have [another simple git ](https://github.com/covrig/homeassistant-hline) that @arsaboo suggested me move to Lovelace. Basically it adds a horizontal line between some entities helping a bit with the grouping/highlighting.

I was thinking to link it to the entities card for the start. Do you see this as a good idea? It should be relatively easy to implement. I was thinking to use an `if` in the `for` loop modifying the css - e.g. border top of the next sensor if the `- horizontal-line` is present. 
There are many ways to accomplish this. Maybe we could allows some options: color, type, thickness etc.

```
  - type: entities
    title: Entities card sample
    entities:
      - input_boolean.switch_ac_kitchen
      - input_boolean.switch_ac_livingroom
      - input_boolean.switch_tv
      - horizontal-line
      - sensor.yweather_pressure
      - sensor.yweather_temperature
```

capture

**Comment by balloob** [comment]
Feature discussions go in https://github.com/home-assistant/ui-schema/issues

**Pull request merged**
